### ISSUE TITLE
support nested JSON?

### ISSUE THREAD

**Issue by hgarc014** [akveo/ng2-smart-table#375]
I have a nested JSON object like:
```
[
    {
        user : {
            "fname" : "Joe",
            "lname" : "Smith"
        },
        execution : {
            ...
        }
    }
...
]
```
I tried methods mentioned in #98 such as:
```
settings:{
        columns:{
              user:{
                     title:"First Name",
                     valuePrepareFunction: (user) => {
                         return user.fname;
                     }
              },
              user:{
                     title:"Last Name",
                     valuePrepareFunction: (user) => {
                         return user.lname;
                     }
              }
        }
}
```
Although this doesn't work and only displays the last one.
![image](https://cloud.githubusercontent.com/assets/5572299/26770169/38759a94-4969-11e7-9c87-4e7f0b3af871.png)

Is there any way to have this functionality? Or do I need to reformat my JSON?

**Comment by hgarc014** [comment]
Looks like there's a pull request that will fix this: #360

**Comment by dgpuranik** [comment]
@hgarc014 you should open this issue. I tried the solution in #360. Its a partial solution.

**Comment by hgarc014** [comment]
Yeah, i tried it as well and it works except it seems to mess up sorting / filtering for the nested object.

**Comment by pg7812** [comment]
Have you fixed nested object filter problem?

**Comment by mathuranubhav** [comment]
The same problem occours when trying to use the Sorting/Filtering on the columns.

**Comment by dgpuranik** [comment]
Any progress? Every enterprise app has nested objects. This feature is very necessary. What need to be done to get this fixed.

**Comment by mariusl-a** [comment]
Anyone got a workaround for this? Version 1.2.2.
I assume a custom component would work, but kinda overkill.. 

Thanks for any tips! :)

**Comment by raycola** [comment]
I resolved like this...

`

// my model
export class Profile {
    id: number;
    desc: string;
}

// my list of values. you can load from services... :-) and map to {value, title} structure
profileFilterList = [
    { value: 1, title: 'A' },
    { value: 2, title: 'B' },
    { value: 3, title: 'C' }
  ];
..
other columns
...
"profile": {
          "title": "Profile",
          valuePrepareFunction: (cell: any, row: any) => { return row.profile.desc },
          filter: {
            type: 'list',
            config: {
              selectText: 'All',
              list: this.profileFilterList
            }
          },
          filterFunction(profile?: any, search?: string | number): boolean {
            
            let match = profile.id == search;
            if (match || search === '') {
              return true;
            } else {
              return false;
            }
          }
         
        },`

Hope it works!
